app/presenters/service_presenters.py

```diff
@@ -0,0 +1,246 @@
+        attribute_groups = []
+        for group in mappings:
+            attribute_group = {
+                'name': group['name'],
+                'rows': self._get_rows(group['rows'], service_data)
+            }
+            if len(attribute_group['rows']) > 0:
+                attribute_groups.append(attribute_group)
+
+        return attribute_groups
+
+    def _get_service_meta(self, service_data):
+        return Meta(service_data)
+
+    def _get_rows(self, group, service_data):
+        rows = []
+
+        for row in group:
+            try:
+                attribute = Attribute(row['value'], service_data)
+            except KeyError:
+                continue
+            data_value = attribute.get_data_value()
+            data_type = attribute.get_data_type(data_value)
+            if data_type is not False:
+                current_row = {
+                    'key': row['key'],
+                    'value': data_value,
+                    'type': data_type
+                }
+
+                rows.append(current_row)
+
+        return rows
+
+
+class Attribute(object):
+    """Wrapper to handle accessing an attribute in service_data"""
+
+    def __init__(self, key, service_data):
+        """Returns if the attribute key points to a row in the service data"""
+        self.key = key
+        self.service_data = service_data
+        self.key_type = self.get_data_type(key)
+        if self.__key_maps_to_data() is False:
+            raise KeyError("Attribute key not found in service data")
+
+    def get_data_type(self, value):
+        """Gets the type of the value parameter"""
+        if self._is_string(value):
+            return 'string'
+        elif isinstance(value, bool):
+            return 'boolean'
+        elif isinstance(value, int):
+            return 'integer'
+        elif isinstance(value, float):
+            return 'float'
+        elif isinstance(value, list):
+            return 'list'
+        elif self._is_function(value):
+            return 'function'
+        elif isinstance(value, dict):
+            return 'dictionary'
+        else:
+            return False
+
+    def get_data_value(self):
+        """Get the value for the attribute key in the service data"""
+        if hasattr(self, 'data_value') is False:
+            if self.key_type is 'function':
+                data_value = self.key(self.service_data)
+            else:
+                data_value = self.service_data[self.key]
+            self.data_value = self.format(data_value)
+            self.data_type = self.get_data_type(data_value)
+        return self.data_value
+
+    def format(self, value):
+        """Formats the value parameter based on its type"""
+        value_format = self.get_data_type(value)
+        if value_format is 'boolean':
+            if value:
+                return u'Yes'
+            else:
+                return u'No'
+        elif (value_format is 'list') and (value[0] == 'None'):
```

Comment:
This line throws an error for the case of empty lists - e.g. `/service/5390159512076288`, `/service/6312248523882496`, `/service/6110806571745280`